src/ripple/app/tx/impl/InvariantCheck.cpp

```diff
@@ -321,12 +321,14 @@ AccountRootsNotDeleted::finalize(
     ReadView const&,
     beast::Journal const& j)
 {
-    // AMM account root can be deleted as the result of AMM withdraw
+    // AMM account root can be deleted as the result of AMM withdraw/delete
     // transaction when the total AMM LP Tokens balance goes to 0.
     // Not every AMM withdraw deletes the AMM account, accountsDeleted_
     // is set if it is deleted.
     if ((tx.getTxnType() == ttACCOUNT_DELETE ||
-         (tx.getTxnType() == ttAMM_WITHDRAW && accountsDeleted_ == 1)) &&
+         ((tx.getTxnType() == ttAMM_WITHDRAW ||
```

Comment:
This is another reason to only let AMMDelete do deletions. The more txn types that are allowed to result in an account root being removed the less secure this invariant is.

Reply:
True, but it would make for a bad user experience. I think in vast majority of the cases there will be way fewer than 1,500 trustlines, especially now that we are not allowing trustlines to AMM unless it's LPTokens. Consequently, the final withdraw is going to be able to delete all trustlines and delete AMM account and object in majority of the cases.